packages/@sanity/cli/src/commands/functions/logsFunctionsCommand.ts

```diff
@@ -1,114 +1,147 @@
+  # Retrieve the last two log entries for Sanity Function
+  sanity functions logs echo --limit 2
 
-  # Retrieve logs for Sanity Function abcd1234 in json format
+  # Retrieve logs for Sanity Function in json format
   sanity functions logs --name echo --json
+
+  # Delete all logs for Sanity Function
+  sanity functions logs --name echo --delete
 `
 
 const defaultFlags = {
-  name: '',
   limit: 50,
   json: false,
   utc: false,
+  delete: false,
+  force: false,
 }
 
 const logsFunctionsCommand: CliCommandDefinition = {
   name: 'logs',
   group: 'functions',
   helpText,
-  signature: '',
-  description: 'Retrieve logs for a Sanity Function',
-  hideFromHelp: true,
+  signature: '<name> [--limit <number>] [--json] [--utc] [--delete [--force]]',
+  description: 'Retrieve or delete logs for a Sanity Function',
   async action(args, context) {
     const {apiClient, output} = context
-    const {print} = output
+    const {print, error: printError} = output
+    const [name] = args.argsWithoutOptions
     const flags = {...defaultFlags, ...args.extOptions}
 
     const client = apiClient({
       requireUser: true,
       requireProject: false,
     })
 
-    if (flags.name === '') {
-      throw new Error('You must provide a function name via the --name flag')
+    if (!name) {
+      throw new Error('You must provide a function name as the first argument')
     }
 
     const token = client.config().token
-    const {blueprint} = await import('@sanity/runtime-cli/actions/blueprints')
+    if (!token) throw new Error('No API token found. Please run `sanity login`.')
+
+    const {getBlueprintAndStack} = await import('@sanity/runtime-cli/actions/blueprints')
     const {findFunction} = await import('@sanity/runtime-cli/utils')
 
-    const {deployedStack} = await blueprint.readBlueprintOnDisk({
-      getStack: true,
-      token,
-    })
+    const {deployedStack} = await getBlueprintAndStack({token})
 
     if (!deployedStack) {
       throw new Error('Stack not found')
     }
 
-    const blueprintConfig = blueprint.readConfigFile()
-    const projectId = blueprintConfig?.projectId
+    const {projectId} = deployedStack
 
-    const {externalId} = findFunction.findFunctionByName(
-      deployedStack,
-      flags.name,
-    ) as StackFunctionResource
+    const {externalId} = findFunction.findFunctionByName(deployedStack, name)
 
     if (token && projectId) {
       const {logs: logsAction} = await import('@sanity/runtime-cli/actions/functions')
-      const {ok, error, logs, total} = await logsAction.logs(
-        externalId,
-        {limit: flags.limit},
-        {token, projectId},
-      )
-
-      if (!ok) {
-        print(`Error: ${error || 'Unknown error'}`)
-        return
-      }
 
-      const filteredLogs = logs.filter(
-        (entry: {level: string; message: string}) => entry.level && entry.message,
-      )
+      if (!externalId) printError('Unable to delete logs. Unable to determine function ID.')
```

Comment:
`printError` throws an exception right? Based on lines 94-95 it doesn't look like it does so we should probably throw here to stop the execution of the command.